### PR TITLE
opentelemetry: embed no-op structs

### PR DESCRIPTION
**Pull request by katiehockman**
### What does this PR do?

Embeds the [opentelemetry-go `noop` structs](https://pkg.go.dev/go.opentelemetry.io/otel/trace@v1.19.1-0.20231019171624-1e1cc901a561/noop) into the DD OTel API. AIT-8692

### Motivation

This is a build requirement for the OTel API after https://github.com/open-telemetry/opentelemetry-go/pull/4620, and our code fails to compile if we upgrade to the latest main without this fix.

See [their package docs](https://pkg.go.dev/go.opentelemetry.io/otel/trace@main#hdr-API_Implementations) for more details.

We had three options that would have 3 different consequences when the upstream makes a breaking change:
- `embedded.TracerProvider` - will break at compile-time
- `trace.TracerProvider` - will build but panic
- `noop.TracerProvider` - will build and any missing functions will be a no-op

The `noop.TracerProvider` sounded like the safest option for us, as it will give us time to support new functions in the API without needing to rush.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [x] There is a benchmark for any new code, or changes to existing code.
- [x] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by katiehockman** [comment]
Note that right now, the tests will fail due to compilation errors in the _upstream_ when we make this change. All of our code builds, but when I try to run our tests, I get errors that look like this:
```
../../../../pkg/mod/go.opentelemetry.io/otel@v1.19.0/internal/global/trace.go:56:30: cannot use &tracerProvider{} (value of type *tracerProvider) as trace.TracerProvider value in variable declaration: *tracerProvider does not implement trace.TracerProvider (missing method tracerProvider)
../../../../pkg/mod/go.opentelemetry.io/otel@v1.19.0/internal/global/trace.go:104:10: cannot use val (variable of type *tracer) as trace.Tracer value in return statement: *tracer does not implement trace.Tracer (missing method tracer)
../../../../pkg/mod/go.opentelemetry.io/otel@v1.19.0/internal/global/trace.go:109:9: cannot use t (variable of type *tracer) as trace.Tracer value in return statement: *tracer does not implement trace.Tracer (missing method tracer)
```
It's unclear to me at first glance if this is something that needs to be fixed on the upstream, or on our side. I'll keep looking into it.

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Overall looks good to me! Maybe we should add a note to the go docs for users of the otel api that if they upgrade their underlying OTEL dependencies for new features before we've built support they will be a no-op?

**Review by darccio — APPROVED** [review]

File: ddtrace/opentelemetry/span.go [hunk 1]
```diff
@@ -19,23 +19,23 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 	otelcodes "go.opentelemetry.io/otel/codes"
 	oteltrace "go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
 )
 
 var _ oteltrace.Span = (*span)(nil)
 
 type span struct {
-	tracer.Span
```
[the remark this answers is not in the document]
Comment (katiehockman) [reply ddtrace/opentelemetry/span.go:27]: Ah yeah I had that originally then deleted it when I was messing around with it. Thanks for the call out.

**Review by ajgajg1134 — APPROVED** [review]

**Comment by katiehockman** [comment]
We will wait to merge this until the upstream OTel repo has released their changes. At that time, I'll update the `go.mod` file to point to the latest published version, merge this, and we'll release ASAP after that.

**Comment by katiehockman** [comment]
The parametric tests are failing now, but they'll work once https://github.com/DataDog/system-tests/pull/1804 is merged.

**Pull request merged**